### PR TITLE
Head motor handlers

### PR DESCRIPTION
**Pull request by sotaoverride**
Add dispatch handlers for a and z motors. Make a and z motors move with the same move group and task. Follow up PR to add unique IDs to both motors, and dedicated tasks.

### CODE REVIEW

File: head/firmware/can_task.cpp [hunk 1]
```diff
@@ -81,6 +81,15 @@ struct motion_controller::HardwareConfig PinConfigurations {
         .port = GPIOC, .pin = GPIO_PIN_4, .active_setting = GPIO_PIN_SET},
 };
 
+struct motion_controller::HardwareConfig PinConfigurations2 {
```
Comment (sfoster1) [inline head/firmware/can_task.cpp:84]: can we give these better names? `pin_configurations_left` and `pin_configurations_right` maybe?

File: head/firmware/can_task.cpp [hunk 2]
```diff
@@ -109,8 +118,26 @@ static motor_class::Motor motor{
     motor_queue,
     complete_queue};
 
+/*z motor*/
+static motor_class::Motor motor2{
```
Comment (sfoster1) [inline head/firmware/can_task.cpp:122]: let's just name this `motor_left` or something, and so on

File: head/firmware/motor_hardware.c [hunk 3]
```diff
@@ -4,9 +4,8 @@ void HAL_SPI_MspInit(SPI_HandleTypeDef* hspi) {
     GPIO_InitTypeDef GPIO_InitStruct = {0};
     if (hspi->Instance == SPI2) {
         /* Peripheral clock enable */
-        __HAL_RCC_SPI2_CLK_ENABLE();
```
Comment (sfoster1) [inline head/firmware/motor_hardware.c:7]: why remove these? where did they go instead?

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Let's fix up some names and figure out these clocks and then this looks good!

File: head/firmware/motor_hardware.c (hunk 3 above)
Comment (sotaoverride) [reply head/firmware/motor_hardware.c:7]: [here](https://github.com/Opentrons/ot3-firmware/blob/main/head/firmware/motor_hardware.c)
```
HAL_StatusTypeDef initialize_spi(SPI_HandleTypeDef* hspi) {
    if (hspi->Instance == SPI2) {
        __HAL_RCC_SPI2_CLK_ENABLE();
    } else if (hspi->Instance == SPI3) {
        __HAL_RCC_SPI3_CLK_ENABLE();
    } else {
        return HAL_ERROR;
    }
    return HAL_SPI_Init(hspi);
}
```

File: head/firmware/can_task.cpp [hunk 4]
```diff
@@ -135,19 +135,19 @@ static motor_class::Motor motor2{
     complete_queue};
 
 /** The parsed message handler */
-static auto can_motor_handler = MotorHandler{message_writer_1, motor};
-static auto can_motor_handler2 = MotorHandler{message_writer_1, motor2};
+static auto can_motor_handler = MotorHandler{message_writer_1, motor_right};
```
Comment (amitlissack) [inline head/firmware/can_task.cpp:138]: Can we use `right` and `left` instead of `2`?

**Review by amitlissack — APPROVED** [review]

**Review by sfoster1 — APPROVED** [review]

**Pull request merged**